### ISSUE TITLE
Thermostats Z-Thrm2fx loosing connections

### ISSUE THREAD

**Issue by HkmeSystem** [TedTolboom/no.ThermoFloor#24]
Hi,
I spent a lot of time on this issue together with Homey support and HeatIT support and we could not figure that out, so it probably is an issue in the app itselve. Please, have a look at it:

1 - adding thermostats was nightmare. It took really long time to add 7 thermostats. In homey, I start the inclussion, then on thermostat I start the inclussion, Homey recognises it and start adding it. On thermostat I get Inc message but then Homey says "no Z-Wave device found" and does not add anything. It takes many many many restores of thermostats, exclussions, reboots... and adding it like 3cm from Homey till it actually work. This is in my opinion definitelly something with the app, as I know the thermostats works nicely with Fibaro, Popp, Raspberry and other control units. As well, it was absolutelly no problem add Aeotec device and fibaro devices on that Homey.

2 - After adding them all, they worked alright. But then I always do power shortage to see how the system recover from it and let me tell you, it did not recover at all :D not even one of those 7 thermostats are working after 1 hour power shortage. Not even the one that is 1 meter from Homey. I only got it to work by pressing buttons on thermostat itselve and basically waking it up... Homey support says it may be because it does not respond to ping from homey after power shortage, but again, if I try it on Fibaro system, after power shortage the thermostats come back to life without any issue. And on homey, all other devices comes back to life without any issue. So that is again pointing at the problem with this app. 

Could you please have a look at it? I already spent a lot of time on it and I can not fiigure it out, so there are expensive thermostats on the wall that are not even controlable remotelly :D

**Comment by HkmeSystem** [comment]
Hi Ted,

thanks for reaching back to me. The comunication was very long and not all that iinformative as we were just trying to find an issue, but I am writing here as we did not find any. As well, I dont have tickets from that as with Athom I comunicated Boaz directly and with Heatit I do communicate directly on emails as well.

But what we figured with Athom - the devices are most times adding as "unknown device" in the developer, but are not visible in app. The steps are completelly the same, but it takes like 10 tries to have it added as thermostat instead of unknown device. Boaz did not find any issue in the system other than that, all is running as it should. He obviously tried all the usual they try, but nothing helped.

With HeatIT it really came only to the conclussion that it is often hard to add those thermostats to Homey and they dont know why. 

About the FW version, I thin those would be 3.6 or max one vesion older. The thermostat were bought from HeatIT 2 months ago. 

Anyway, if it would be any help to you, I can forward you the access to that homey and you can dig around yourselve. Maybe you can find something else there. If that helps, please send me email to milos.bucek@homesystem.sk I will provide you the credentials. Thanks

**Comment by TedTolboom** [comment]
Hi Milos,

Thanks for your fast response.
In the meantime, I also received messages from both Mats and Boaz...
I'm trying to reconstruct all information provided to get a clear picture of the issue.

Providing access to that Homey does not provide any additional value without also having a VPN connection to the local network Homey is on. Otherwise I will not have access to the debug logging. And it will still require someone in addition to push the buttons on the thermostat...

Instead, I will try to find some time early next week to reproduce the issue at my development setup (after installing FW 3.6 on it) and see if I can reproduce the issue experienced.

**Comment by HkmeSystem** [comment]
Oh tha ks so much. I am on holiday now, but next tuesday i will be there and i can provode you with teamviewer or whatever you would need. So if ypu would not be able to reproduce that i am more than happy to assist you. As well, I can donate something for your time...On Sep 13, 2019 20:22, Ted Tolboom <notifications@github.com> wrote:Hi Milos,
Thanks for your fast response.
In the meantime, I also received messages from both Mats and Boaz...
I'm trying to reconstruct all information provided to get a clear picture of the issue.
Providing access to that Homey does not provide any additional value without also having a VPN connection to the local network Homey is on. Otherwise I will not have access to the debug logging. And it will still require someone in addition to push the buttons on the thermostat...
Instead, I will try to find some time early next week to reproduce the issue at my development setup (after installing FW 3.6 on it) and see if I can reproduce the issue experienced.

**Comment by TedTolboom** [comment]
Hi Milos,

I’ve spend today quite some hours to try to reproduce the issues with the Z-TRM2fx as reported by you.

I’ve reconnected my development Z-TRM2fx thermostat (on FW 3.4) after several weeks being disconnected and it connected without issues and started to work as expected.

I’ve updated my Z-TRM2fx to FW 3.6 (or 3.06)

- After updating the thermostat, I re-included the thermostat at 5 meter range of Homey (same room, on my private Homey with active and extensive network) which failed (time-out)
- The thermostat had completed the inclusion successfully and Homey didn't
- Removing the thermostat from Homey resulted in a “Err” on the thermostat display but a message that the device was successfully removed.
- Reset the thermostat based on the described manual reset procedure (without reset the thermostat will not properly include): By pressing buttons Right and Center (down and confirm) for 20 seconds, the thermostat will perform a complete factory reset.
- Moved Homey to the proximity thermostat: preferred method is to connect Homey on a strong enough power bank. Wait until Homey is completely ready and has booted all apps / protocols (~30 minutes).
- Started the inclusion proces again in close proximity and completed successfully
- After successful inclusion, I validated all functions of the thermostat and driver and all was working ok without any modifications to the association groups / parameters
- Homey UI updating based on changing mode / setpoint on the thermostat. As well as deliberately triggering the temperature sensor.
- Thermostat updating correctly based on changing mode / setpoint in the Homey UI

In addition, I tried to reproduce the issues you reported:
- Pulled the power cable of the thermostat and waited for several minutes before reconnecting it: the thermostat reconnected without issues
- Pulled the power cable of the thermostat and waited for half an hour and an hour: the thermostat reconnected without issues
- Pulled the power cable of the thermostat and Homey waited for hal an hour before reconnecting both: the thermostat reconnected without issues
- Pulled the power cable of the thermostat and waited for half an hour during which I send a command from Homey (different setpoint / mode which all timed out). The device is marked as “Unreachable” by Homey:
<img width="1170" alt="Schermafbeelding 2019-09-14 om 20 35 41" src="https://user-images.githubusercontent.com/17163431/64912861-9be18f00-d735-11e9-83ff-b40f9485722e.png">

Reconnected power to the thermostat and the device recovered. So I can’t reproduce the issue on my development thermostat

Thinking out loud:
- Considering the uptime of Homey (~ 2.5) hours, there is an huge amount of communication between thermostat and Homey (1688 Tx and 2862 Rx), which points to security handshake issues… Note: my Greenwave power node devices are at a 300 level...
- When this amount of communication is originating from 1 thermostat, the traffic of 7 thermostats might overload the network / Z-wave chip of Homey…

So I would draw the conclusion, without being able to reproduce it, that the S2 security handshake issues lead to additional traffic on the Network and with 7 of those devices, it will overload the network / Z-wave chip. So I think this is the culprit. I can’t judge where the security handshake issues are originating from. Yes, Homey is formally not S2 compatible, but the thermostat should also be able to work based on S0 security.

I am currently in contact with Athom to check what the timeline is to implement S2 security on Homey; which could resolve the issues. As well as in contact with Heatit to share these learnings.

In addition, I will check if another solution is possible. Keep you updated.

**Comment by HkmeSystem** [comment]
Hi,I can not thank you enough for your effort.Please do you think then downgrading the thermostats to 3.4 could fix the issues on that instalation?I see only one difference there - the network I have issue at is builded only by 7 thermostats, 1 aeotec relay and 2 fibaro sensors. So the mesh is basically dependand on the thermostats. Or do you have any other sugestions I could try. I still will be on the place of instalation in a week so I still can give you the teamviewer access to it. Actually i can try to schedule it with my client any other day as well, but at the evening only. Thanks againOn Sep 14, 2019 21:25, Ted Tolboom <notifications@github.com> wrote:Hi Milos,

I’ve spend today quite some hours to try to reproduce the issues with the Z-TRM2fx as reported by you.

I’ve reconnected my development Z-TRM2fx thermostat (on FW 3.4) after several weeks being disconnected and it connected without issues and started to work as expected.

I’ve updated my Z-TRM2fx to FW 3.6 (or 3.06)

After updating the thermostat, I re-included the thermostat at 5 meter range of Homey (same room, on my private Homey with active and extensive network) which failed (time-out)The thermostat had completed the inclusion successfully and Homey didn'tRemoving the thermostat from Homey resulted in a “Err” on the thermostat display but a message that the device was successfully removed.Reset the thermostat based on the described manual reset procedure (without reset the thermostat will not properly include): By pressing buttons Right and Center (down and confirm) for 20 seconds, the thermostat will perform a complete factory reset.Moved Homey to the proximity thermostat: preferred method is to connect Homey on a strong enough power bank. Wait until Homey is completely ready and has booted all apps / protocols (~30 minutes).Started the inclusion proces again in close proximity and completed successfullyAfter successful inclusion, I validated all functions of the thermostat and driver and all was working ok without any modifications to the association groups / parametersHomey UI updating based on changing mode / setpoint on the thermostat. As well as deliberately triggering the temperature sensor.Thermostat updating correctly based on changing mode / setpoint in the Homey UI

In addition, I tried to reproduce the issues you reported:

Pulled the power cable of the thermostat and waited for several minutes before reconnecting it: the thermostat reconnected without issuesPulled the power cable of the thermostat and waited for half an hour and an hour: the thermostat reconnected without issuesPulled the power cable of the thermostat and Homey waited for hal an hour before reconnecting both: the thermostat reconnected without issuesPulled the power cable of the thermostat and waited for half an hour during which I send a command from Homey (different setpoint / mode which all timed out). The device is marked as “Unreachable” by Homey:



Reconnected power to the thermostat and the device recovered. So I can’t reproduce the issue on my development thermostat

Thinking out loud:

Considering the uptime of Homey (~ 2.5) hours, there is an huge amount of communication between thermostat and Homey (1688 Tx and 2862 Rx), which points to security handshake issues… Note: my Greenwave power node devices are at a 300 level...When this amount of communication is originating from 1 thermostat, the traffic of 7 thermostats might overload the network / Z-wave chip of Homey…

So I would draw the conclusion, without being able to reproduce it, that the S2 security handshake issues lead to additional traffic on the Network and with 7 of those devices, it will overload the network / Z-wave chip. So I think this is the culprit. I can’t judge where the security handshake issues are originating from. Yes, Homey is formally not S2 compatible, but the thermostat should also be able to work based on S0 security.

I am currently in contact with Athom to check what the timeline is to implement S2 security on Homey; which could resolve the issues. As well as in contact with Heatit to share these learnings.

In addition, I will check if another solution is possible. Keep you updated.

**Comment by TedTolboom** [comment]
Hi Milos,
No, downgrading to 3.4 does not solve the issue since also 3.4 is using S2 security.

Best option for now, until Homey if fully S2 compatible, would be to force non-secure inclusion of the Thermostat to Homey. From a principal point of view this is not my preferred option; but the only one working reliably.

Would that be an acceptable solution; from your reseller / installer point of view?

I am able to force (not optional) non-secure inclusion from driver side and am testing this driver as I type on my development thermostat. So far, working fine with the exception of an `Err` (instead of `Incl`) showing on the device after inclusion. But this is not hampering the working of the device. Suspect that this is a FW issue at Heatit side; have seen more reports on other platforms.

Before releasing it as app, I’d like to first await feedback of Heatit and Athom

**Comment by HkmeSystem** [comment]
Well, if homey does not support s2 it is actually worthless in thermostat itselve right? And it is beter to have thermostat with s0 working then with s2 non working. I will probably go to athom company on tuesday so i will try to talk to them about it a bit. But if you even can release a beta with s0 that would be probably the option for now, as everyone need to apply for beta and probably only people who has this isssue would, right?On Sep 15, 2019 12:41, Ted Tolboom <notifications@github.com> wrote:Hi Milos,

No, downgrading to 3.4 does not solve the issue since also 3.4 is using S2 security.

Best option for now, until Homey if fully S2 compatible, would be to force non-secure inclusion of the Thermostat to Homey. From a principal point of view this is not my preferred option; but the only one working reliably.

Would that be an acceptable solution; from your reseller / installer point of view?

I am able to force (not optional) non-secure inclusion from driver side and am testing this driver as I type on my development thermostat. So far, working fine with the exception of an Err (instead of Incl) showing on the device after inclusion. But this is not hampering the working of the device. Suspect that this is a FW issue at Heatit side; have seen more reports on other platforms.

Before releasing it as app, I’d like to first await feedback of Heatit and Athom

**Comment by TedTolboom** [comment]
Hi Milos,

Agree to that.
I expect to be able to have the updated app available as separate branch on Github either today or tomorrow. Pending on the feedback of Heatit and Athom, I will release it as BETA branch to the app store.

Both Emile and Boaz (from Athom), as well as Mats (from Heatit) are in detail informed on the data I collected and insights gained. Not all data I can publicly share on GH.

**Comment by TedTolboom** [comment]
The app update that will force non-secure inclusion is available at: https://github.com/TedTolboom/no.ThermoFloor/tree/fix/force-non-secure-inclusion

One of the active users will test it tomorrow evening on a multiple thermostat setup

**Comment by HkmeSystem** [comment]
That was very fast, thanks. Does he have the issues with thermostats?On Sep 15, 2019 21:31, Ted Tolboom <notifications@github.com> wrote:The app update that will force non-secure inclusion is available at: https://github.com/TedTolboom/no.ThermoFloor/tree/fix/force-non-secure-inclusion
One of the active users will test it tomorrow evening on a multiple thermostat setup

**Comment by TedTolboom** [comment]
Yes, thermostats that stop reporting / being able to be controlled

**Comment by TedTolboom** [comment]
@HkmeSystem, tests are looking positive at my development thermostat and initial feedback is also positive from the active user installing the update at his home (~7x Z-TRM2fx).

I'm awaiting final positive feedback, but expect to release v2.3.0 (BETA) to the app store by tomorrow. It is already available at: https://github.com/TedTolboom/no.ThermoFloor/tree/beta

**Comment by TedTolboom** [comment]
@HkmeSystem v2.3.0-BETA has been uploaded to the Homey Apps store to the BETA branch of the Heatit app, awaiting Athom's approval.

Also in the tests at the active user, it proved to be a huge improvement in reliability, stability and response time.

I'll ask Emile tomorrow to approve this release; hopefully in time to finalize your project.

Kind regards,

Ted

**Comment by HkmeSystem** [comment]
Ted thanks a lot, I talked to Boaz about it on Monday, he was as well telling me it may be the S2, so I hope it really fixes it. I will test it on the instalation and I will keep you updated.

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )

**Comment by TedTolboom** [comment]
Hi Milos,

The Beta has been released to the app store this morning: https://apps.athom.com/app/no.thermofloor

Looking forward to your feedback.

Ted

**Comment by HkmeSystem** [comment]
Ted I wanted to ask you one more question. We are trying the guy behind TKB Home app to add one device (thermostat) but he said he is not developing it anymore but if someone would write the support for the thermostat he would add it the the app.So I wanted to ask, would you be up for that? I would send you the thermostat and as well we can pay something for your time. Please let me know if you would consider. Thanks.On Sep 19, 2019 10:18, Ted Tolboom <notifications@github.com> wrote:Hi Milos,
The Beta has been released to the app store this morning: https://apps.athom.com/app/no.thermofloor
Looking forward to your feedback.
Ted

**Comment by TedTolboom** [comment]
Hi Milos,
Did you have an opportunity to install the Beta at your projects location?
Does it resolve the issue with the 7 Z-TRM2fx's and can we close this issue as being resolved?

Looking forward to your feedback.
Kind regards,

Ted

**Comment by HkmeSystem** [comment]
Hi,Sorry but no, I was traveling this week for some meetings, but the installation is scheduled on Tuesday so I will know more. I will keep you informed, don't worryOn Sep 21, 2019 17:19, Ted Tolboom <notifications@github.com> wrote:Hi Milos,
Did you have an opportunity to install the Beta at your projects location?
Does it resolve the issue with the 7 Z-TRM2fx's and can we close this issue as being resolved?
Looking forward to your feedback.
Kind regards,
Ted

**Comment by HkmeSystem** [comment]
Hi Ted,

thanks again for helping with the issue with thermostats.
I went today to my customers house to try to fix them. I reinstalled them all on beta. It did not go without issues, few times they did not add but at the end it worked.

But I tested as well the power cut situation and sadly, even slightest power cut caused the thermostats to stop comunicating with Homey. The MESH gets broken, last known roads disapears and whatever I tried doing with the app, the thermostat were not controlable. The only thing that helped was playing with the thermostats themselves, changing the temp... eventually got them back to life. So at the end it works, but I would call it really fragile :)

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )

**Comment by TedTolboom** [comment]
Hi Milos,

One step to clarify. When re-adding them to Homey, which option did you use?
“Add device > Heatit > Z-TRM2fx” followed by the detailed inclusion instructions or “Add device > Homey > Z-wave” followed by the generic inclusion instructions?
The first will add the thermstat as non-secure, the latter will result in the old, secure way…. could you verify if the thermostats are added as non-secure?

In the forum topic, I’ve added a detailed inclusion instructions:
How to re-include the Z-TRM2fx non-secure to Homey

Note: similar approach can be followed for the Z-Water and Z-relay

Install v2.3.0 BETA version of the app
Move Homey to close proximity of the thermostat (Homey does not support inclusion of the network) and wait sufficient time before Homey finished starting up all apps (~ 30 minutes.
If multiple thermostats need to be re-included, connect Homey to a powerfull enough powerbank and move it around.
Remove the thermostat from Homey, by going to ‘device settings’ > ‘Remove device’ and follow the instructions (mind updated step 5):
Press the Center (confirm) button for 10 seconds. The display will show OFF.
Press Right(down) button 5 times till you see Con in the display.
Now start to remove the device from Homey.
Start exclusion mode by pressing Center(confirm) button for approximately 2 seconds. The inclusion / exclusion icon will appear in the display.
Confirmation will show EcL in the display.
When Err(error) appears, check if the thermostat has been removed from Homey. If so, reset the thermostat according to the instructions of the manual. Otherwise repeat the exclusion.
After removal from Homey, manually reset the thermostat by pressing the right 2 buttons of the thermostat for 20 second, during the last few seconds res will be shown in the display.
Set Homey in inclusion mode by only using the following method Add device > Heatit > Z-TRM2fx , the specific Z-TRM2fx inclusion instructions should be shown
Don’t use the generic option Add device > Homey > Z-wave , since it will not overrule the default secure inclusion!
Re-include the thermostat to Homey by following the inclusion instructions (mind the addition in step 5):
press the Center (confirm) button for 10 seconds. The display will show OFF.
Press Right(down) button 5 times till you see Con in the display.
Now start add a device to Homey.
Start inclusion mode by pressing Center(confirm) button for approximately 2 seconds. The inclusion / exclusion icon will appear in the display.
Confirmation will show Inc in the display.
If Err (error) is shown on the thermostat display, inclusion might still be successfull; check the functioning of the newly added device.

On your power cut situation. Can you describe what you do exactly? Are both Homey and the Thermostats cut from the power and restored at the same time?

Kind regards,

Ted

> Op 24 sep. 2019, om 21:23 heeft HkmeSystem <notifications@github.com> het volgende geschreven:
> 
> Hi Ted,
> 
> thanks again for helping with the issue with thermostats.
> I went today to my customers house to try to fix them. I reinstalled them all on beta. It did not go without issues, few times they did not add but at the end it worked.
> 
> But I tested as well the power cut situation and sadly, even slightest power cut caused the thermostats to stop comunicating with Homey. The MESH gets broken, last known roads disapears and whatever I tried doing with the app, the thermostat were not controlable. The only thing that helped was playing with the thermostats themselves, changing the temp... eventually got them back to life. So at the end it works, but I would call it really fragile :)
> 
> V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.
> 
> HomeSystem s.r.o.
> 
> www.HomeSystem.sk ( http://www.homesystem.sk/ )
> 
> *email:* info@homesystem.sk
>  
> *Všetky novinky a recenzie nájdete aj na :*
> 
> Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )
>

**Comment by HkmeSystem** [comment]
Hi Ted,
thanks for fast answer.

I added them as Heatit Z-Therm2fx. Basically I did all mentioned, only I did not do the thermostat reset, did not occure to me as I did have just few issues. But yes, the thermostats are added in non secure way, I did check that.

And with the power cut, I did cut only the power of thermostats. I know this may be a bit different if I would cut power from homey too, but still, even the slightest power cut makes such mess :/

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )

**Comment by TedTolboom** [comment]
Hi Milos,

I'm trying to reproduce the issues with power cutting, but fail miserably (apparently)...
In what ever way I unplug my thermostat... after replugging the thermostat, it starts reporting again without interaction needed. But this is a single device in a strong Z-wave network that is not unplugged.

In your case, if I understand correctly, the entire network consists of 7 thermostats and a sensor. Correct? If all 7 mesh points disappear due to a power cut, the recovery of the mesh network might cause the problems; specifically if some of the nodes are dependent of other nodes.

I honestly don't know what further to do to improve this.
The Z-wave mesh network(ing) is not something I can control or influence from the app (for security reasons); this is handled through the Z-wave core of Homey. 

I'll check with Athom (Boaz) if he has some idea...

**Comment by TedTolboom** [comment]
But just to check that I completely understand the issue; the power shutdown is a functional test to test the recovery after an incidental power outage?

**Comment by HkmeSystem** [comment]
Hi, Yes I only tested it. It should never happen like that by itselve.

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )

**Comment by HkmeSystem** [comment]
Yes, almost as you said. There are two motion sensors and now two additional relay switches. The relay switches wake up without issue. Thermostats has an issues. But I know this may be based on that mesh network, as really basically everything depends on the thermostats. But I think it is already working the best way possible, for now the thermsotats are responsive... 
So really thank you for your help. I wanted to ask you, if I do the donation on the app itselve, will there be any invoice or anything issued? I would like to pay you something for your help but I will do it from company account so I need some tax document for it.

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )

**Comment by TedTolboom** [comment]
Hi Milos, see my comment by mail.

If you agree, I will close this issue.

**Comment by HkmeSystem** [comment]
Yes, thanks a lot.

V prípade akýchkoľvek ďalších otázok nás neváhajte kontaktovať.

HomeSystem s.r.o.

www.HomeSystem.sk ( http://www.homesystem.sk/ )

*email:* info@homesystem.sk
 
*Všetky novinky a recenzie nájdete aj na :*

Facebooku ( https://www.facebook.com/homesystem.sk/ ) a YouTube ( https://www.youtube.com/channel/UCVOWBEMYDcDOYo_2L2zHrqA )
